### ISSUE TITLE
Notifications VC: New Works by Artist You Follow (Bell)

### ISSUE THREAD

**Issue by katarinabatina** [artsy/eigen#1158]
This design does not differ too greatly from what is currently served up by the microgravity view accessed through the YOU VC but in order to increase reliability of the view coming from push it makes sense to make this change. 

Included are specs for the view across device types and corresponding zero states. The zero state of this view could have a greatly improved user experience but in the spirit of keeping the scope small we'll stick with this for now. 

![image](https://cloud.githubusercontent.com/assets/6482926/13061215/9f36ac4a-d404-11e5-844b-082b323ad7d9.png)

-----

DEV TODO:

- [x] Switch for native / web
- [x] Basic view - almost there now
- [x] Empty view
- [x] Network model
- [x] Mark as read functionality
- [x] Snapshot tests for VC
- [x] Tests for network model
- [x] Disable rotation
- [x] Fix padding on embedded model VC
- [x] iPad padding and spacing 
- [x] Artist linking

**Comment by katarinabatina** [comment]
cc: @mennenia @sarahscott, sounds like this might be a nicely scoped project for the two of you to pair on, let me know if you have any questions!

**Comment by katarinabatina** [comment]
cc: @alloy I know we talked about pre-loading this view for push notifications, don't know if there's anything we can do upfront in the implementation to make that "easier"

**Comment by katarinabatina** [comment]
cc: @anipetrov I know we recently created a few issues around implementing better analytics for this view. Would you mind giving addressing which analytics events we'll want to track from this view?

The three I can think of are...
[-] artwork tapped
[-] tapped YOU
[-] viewed YOU

**Comment by alloy** [comment]
This work also includes keeping track of the unread count badge on the tab and app icon, which is comprised of the following while scrolling through the new works:

- Update the read status of works on the API
- Update the count badge on the bell tab and app icon

**Comment by orta** [comment]
Hrm, FWIW, those sections of masonry layouts may be tricky to do efficiently. _Probably_​ best to do it dumbly, and include AREmbeddedModelVCs in an ORStackScrollView rather than try to be smart about cell reuse in a tableview here. You’ll still have some tricky AL stuff, but it won’t be anywhere near as difficult as trying to do it with memory as a constraint.

**Comment by mennenia** [comment]
So, silly question - at the bottom there are two tabs on the right "YOU" and the little bell icon. 

The "you" page goes to a page with a switchview at the top: works | artists | categories
The bell goes to a "works by artists you follow", which looks like the same content we've got in this issue.

Which of the two tabs are we actually modifying? You or the notification bell?

**Comment by alloy** [comment]
The bell tab, which shows what we call “works-for-you”, e.g. https://www.artsy.net/works-for-you

**Comment by mennenia** [comment]
Cool, that's what we figured, but it had also been referred to as the "you" page, so just double checking :) :+1:

**Comment by alloy** [comment]
Oh you’re right, the issue title should be fixed.

**Comment by anipetrov** [comment]
@katarinabatina as far as I understand

 
- [ ] we can instrument `works for you artwork tapped` 
- [ ] we have screenviews which reflect landing on a screen for each of the main nav items, however it will be good to create tap events for each: `You tapped`, `Shows tapped`, `Home tapped`, `Mag tapped` the way [Orta instrumented `Bell tapped`](https://github.com/artsy/eigen/pull/1126)
- [ ] `viewed YOU` should be currently instrumented as a screenview named `You` with property `tab` (whether you are landed on artists, artworks, categories)
- [ ] we should instrument a `Bell` or `works for you` screenview and think what properties (if any) to set for it, eg a bool `bell state` or `new works available` (whether the bell had purple number notification upon clicking or not) etc.

**Comment by katarinabatina** [comment]
@mennenia sorry for the confusion it is indeed the notifications bell tab, (we're renaming it to FOR  YOU on the web so I got my wires crossed)

**Comment by mennenia** [comment]
@orta so this is what we were thinking (with a stackview between the artists title and the masonry layout of their artworks)

Happyyy to take feedback/suggestions before we start with this? 

![ouridea](https://cloud.githubusercontent.com/assets/373860/13082072/ffede19e-d4c6-11e5-975b-1f6b6602e423.png)

**Comment by mennenia** [comment]
Was the thing you suggested the following?

Have an AREmbeddedModelsVC for the artworks (instead of masonry), and then stack them with a stackview rather than chucking them in a tableview?

**Comment by orta** [comment]
The `AREmbeddedModelsVC` is how we do masonry stuff, I was thinking that using a tableview to host the cells might end up more tricky than you anticipate, and that given there is a non-infinite set of recommendations, it would probably be easier to ditch the tableview entirely

**Comment by mennenia** [comment]
:+1:

**Comment by orta** [comment]
:+1:

**Comment by mennenia** [comment]
So sorry to be asking more questions... but what is the purpose of this view? This sounds silly but bear with me! :D

![questions](https://cloud.githubusercontent.com/assets/373860/13109580/10a6e778-d572-11e5-990a-d037a97c219f.png)

The way we seem to be talking about it (at least in #mobile  and from what I can see from Gravity in me/notifications/feed), it seems to be Option A - a view only showing you the new things. 

The design above though (specifically the empty state) and the current webview of Works For You, seem like it's about Option B. (which means we can't just use the feed)

Also, I'm assuming that order of the list will be by date? Latest date that something was added to that artist? This is just a sanity check :).

**Comment by katarinabatina** [comment]
@mennenia first and foremost, questions are fantastic never apologize, besides, you asking them probably means I did not provide enough info, like in this case :)

The bell as an icon is a bit misleading, it suggests that Option A would be true but indeed Option B is the correct implementation, a full record of all works that have been uploaded by artists you follow, ordered by date. 

Here is a scroll through my feed this morning. http://cl.ly/3M2o0r3g2a2D I realize I did not mention the read state of each batch of works. We seemed to have abandoned that behavior on the web and on martsy. I think it's a "nice to have" at best, do you think it would be tricky to implement? If not, here's an updated spec http://cl.ly/453l2b372s2c

The only time the user would see the zero state is if they have never followed an artist, once they follow an artist, we backfill the upload history of that artist (last 10 works uploaded) You can see an old PR here from when the web implemented this back-fill logic. https://github.com/artsy/gravity/blob/1820362940e242d682d0c99a3ef18fdea23e3ea2/app/api/v1/artists_artworks_endpoint.rb#L14

**Comment by katarinabatina** [comment]
@mennenia one additional open question I have. We currently display all artworks uploaded (as does the web), we've talked about limiting this to works that are only for sale. This has a potential impact on the frequency of push notifications for a user @alloy would it require changes to the code for push?

**Comment by mennenia** [comment]
In terms of displaying only the works that are for sale, that should be fine to do because we can filter the feed on that and only display works for sale. 

I have no clue about the changes for push unfortunately though :(.

**Comment by alloy** [comment]
>  This has a potential impact on the frequency of push notifications for a user @alloy would it require changes to the code for push?

@katarinabatina Yup.

**Comment by alloy** [comment]
@katarinabatina To be clear, that push notification work needs to happen in Gravity. If/when those changes need to be made, please file a separate ticket for it.

**Comment by mennenia** [comment]
TODO:

- [x] Switch for native / web
- [ ] Basic view - almost there now
- [ ] Empty view
- [x] Network model
- [ ] Mark as read functionality
- [ ] Snapshot tests for VC
- [ ] Tests for network model

**Comment by alloy** [comment]
@mennenia I’d prefer those TODO lists to go into the original issue at the very top (and maybe add a divider between the list and the original description).

**Comment by katarinabatina** [comment]
@sarahscott not sure if you're tackling this issue today but one thing I noticed is that the search button doesn't disappear as I scroll like it does in other places in the app, it should continue that behavior on this view as well.

**Comment by sarahscott** [comment]
@katarinabatina sounds good. The WorksForYouVC in the beta is a first draft - I'm working on the final details now!

**Comment by katarinabatina** [comment]
@sarahscott cool! looking forward to checking it out lemmie know if you wanna do in person QA since YOU'RE BACKKKKKK!

**Comment by sarahscott** [comment]
The given text sizes look a little weird to me in the empty state and could be changed, but other than that, I think WorksForYou is QA and possibly release-ready :dancer:

**Comment by alloy** [comment]
This has been merged and enabled.
